lapis2-docs/src/components/ConfigGenerator/MetadataWizard.tsx

```diff
@@ -0,0 +1,219 @@
+        }
+        if (metadata.type === 'date' && config.dateToSortBy === metadata.name) {
+            removeConfigField('dateToSortBy');
+        }
+        if (metadata.type === 'pango_lineage' && config.partitionBy === metadata.name) {
+            removeConfigField('partitionBy');
+        }
+        deleteMetadata(index);
+    }
+
+    return (
+        <div className='flex flex-col mb-4 min-w-full'>
+            <h1 className='text-xl font-bold mb-4'>Metadata</h1>
+            <div className='card bg-base-100 shadow-xl'>
+                <table className='table '>
+                    <thead>
+                        <tr>
+                            <th>Name</th>
+                            <th>Type</th>
+                            <th>Generate Index</th>
+                            <th></th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {config.metadata.map((metadata, index) => {
+                            const modal = `modal_${metadata.name.replace(' ', '')}`;
+
+                            return (
+                                <tr key={metadata.name}>
+                                    <td>{metadata.name}</td>
+                                    <td>{metadata.type}</td>
+                                    <td>{metadata.generateIndex ? 'yes' : 'no'}</td>
+                                    <td>
+                                        <div className='join'>
+                                            <button
+                                                className='btn'
+                                                onClick={() => {
+                                                    (document?.getElementById(modal) as HTMLFormElement)?.showModal();
+                                                }}
+                                            >
+                                                <EditIcon />
+                                            </button>
+                                            <button
+                                                className='btn'
+                                                onClick={() => handleDeleteMetadata(metadata, index)}
+                                            >
+                                                <DeleteIcon />
+                                            </button>
+                                        </div>
+
+                                        <MetadataEditModal index={index} metadata={metadata} modal={modal} />
+                                    </td>
+                                </tr>
+                            );
+                        })}
+                    </tbody>
+                </table>
+                <div className={'flex justify-end'}>
+                    <button
+                        className='btn'
+                        onClick={() => {
+                            addNewMetadata();
+                        }}
```

Comment:
How about directly opening the modal when clicking this? It should be a rather rare case that you don't want to edit the default (given that the default name is rather useless).

Or opening a separate modal that only adds the new field upon clicking "Ok" in that modal?

Reply:
I added a new ticket for this feature: #617